### PR TITLE
Export Public Key and Sign Message #1155

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1155 
add two commands:
1、ioctl account exportpublic fist
Enter password #fist:
04d4f7be66bf121f0d1dfe0a4ebe30561d40d71932994d2fbc86fd4ef25e71b8923e5d39787f0a44a1c5574a7b8c5b8c4dcecea5c1be5a2b0c040c8d15a93d743b
2、ioctl account sign fist 0xdeadbeefdafa
Enter password #fist:
8aab5952493d0a5e7436115a226d6837a6f1b07f46cb49764194a7e12dd428107e8f877ec7c2f5bffa2f7c02d5a8c95e2ecb87be894314bc30df5d3cae6fba7500

msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(message), message)
signed msg is same as ethereum

### CODE REVIEW

File: cli/ioctl/cmd/account/account.go [hunk 1]
```diff
@@ -9,24 +9,26 @@ package account
...
+	"encoding/hex"
 	"fmt"
 	"syscall"
 
 	"github.com/ethereum/go-ethereum/accounts/keystore"
-	"github.com/iotexproject/go-pkgs/crypto"
-	"github.com/iotexproject/iotex-address/address"
-	"github.com/iotexproject/iotex-proto/golang/iotexapi"
-	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
```
Comment (dustinxie) [inline cli/ioctl/cmd/account/account.go]: use go-pkgs/crypto for message signing
we don't want to introduce new dependency here

File: cli/ioctl/cmd/account/account.go [hunk 2]
```diff
@@ -9,24 +9,26 @@ package account
...
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"go.uber.org/zap"
 	"golang.org/x/crypto/ssh/terminal"
 	"google.golang.org/grpc/status"
 
+	"github.com/iotexproject/go-pkgs/crypto"
```
Comment (dustinxie) [inline cli/ioctl/cmd/account/account.go]: change back to original import style, iotex-address, iotex-proto do not belong to iotex-core internal project

File: cli/ioctl/cmd/account/account.go [hunk 3]
```diff
@@ -47,16 +49,34 @@ func init() {
...
 }
 
+// Sign sign message with signer
+func Sign(signer, password, message string) (signedMessage string, err error) {
+	pri, err := KsAccountToPrivateKey(signer, password)
+	if err != nil {
+		return
+	}
+	msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(message), message)
+	mes := ethcrypto.Keccak256([]byte(msg))
```
Comment (dustinxie) [inline cli/ioctl/cmd/account/account.go]: use go-pkgs/hash.Hash256()

Comment (lzxm160) [reply cli/ioctl/cmd/account/account.go]: sure

File: cli/ioctl/cmd/account/account.go [hunk 4]
```diff
@@ -47,16 +49,34 @@ func init() {
...
 		"insecure connection for once")
 }
 
+// Sign sign message with signer
+func Sign(signer, password, message string) (signedMessage string, err error) {
+	pri, err := KsAccountToPrivateKey(signer, password)
+	if err != nil {
+		return
+	}
+	msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(message), message)
```
Comment (dustinxie) [inline cli/ioctl/cmd/account/account.go]: did some study, looks like we are trying to implement web3.eth_sign() here, so our user can use iotex address to sign eth message

https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign
it is taking message to sign in hexstring format, "0xdeadbeef"

so we need to interpret 'message' as a hex string, same as we discussed in antenna-go project, this also allows user to sign a message that is not necessarily a text string

b, err := hex.DecodeString(message)
if err != nil {
    return
}
prefix := fmt.Sprintf("\x19Ethereum Signed Message:\n%d", len(b))
msg := append([]byte(prefix), b...)

Comment (lzxm160) [reply cli/ioctl/cmd/account/account.go]: ok

**Pull request merged**